### DISCUSSION TITLE
How do I handle cases where error doesn't match the expected format?

### DISCUSSION THREAD

**Discussion by haggs** [openapi-ts/openapi-typescript#1395] [Q&A]
Hi, I couldn't figure out what to do from reading the docs so here I am. 

First off, I love using this tool. I'm using it with TanStack Query and it's going very well so far.

However, sometimes, due to unknown circumstances, occasionally I get a response that doesn't adhere to my OpenAPI spec, and a client.GET call will throw.

Is there something I can configure to have a default response? I noticed `default` response mentioned in the docs but couldn't figure out where to set that up.

**Comment by drwpow** [comment]
`default` response is usually for errors, and you specify it rather than a status code in your OpenAPI schema, e.g.:

```yaml
responses:
  200:
    # …
  default:
    # …
```

But `openapi-fetch` will always prefer status codes if they are specified, and will only use `default` for the error type if another error code (e.g. `500`) isn’t given. So even if you added a `default` type, it would still probably give you the types that your OpenAPI schema is telling you it’ll get (which should never change).

I can’t say for sure, but it sounds like your endpoint is throwing errors while returning “OK” status codes, which `openapi-fetch` can’t really account for since it’s all static type analysis with zero runtime.

**Reply by haggs** [reply]
Got it. I'm actually already using that, but my `default` definition has a JSON response body, which isn't always true. I guess most people have a more generic default response than that.

**Reply by haggs** [reply]
It's not that my endpoint is returning 200, it's that it's *not* returning a JSON body that i've defined in `default` in very specific cases, i.e. if the backend is completely unreachable.

**Reply by haggs** [reply]
I think the answer here is to have my OpenAPI definition account for failures where you don't get a response body, so my openapi-fetch client won't throw.
